db/crud.go

```diff
@@ -1712,13 +1713,19 @@ func (db *DatabaseContext) assignSequence(ctx context.Context, docSequence uint6
 		// so we're allowing more 'recent sequences' on the doc (20) before attempting pruning
 		stableSequence := db.changeCache.GetStableSequence(doc.ID).Seq
 		count := 0
+		// we want to keep at least kMinRecentSequences recent sequences in the recent sequences list to reduce likelihood
+		// races between compaction of resent sequences and a coalesced DCP mutation resulting in skipped/abandoned sequences
+		maxToCompact := len(doc.RecentSequences) - kMinRecentSequences
```

Comment:
This number can be negative, in which case we will end up compacting all sequences. Is it possible to write a unit test which would hit this and make sure that this function is robust to `len(doc.RecentSequences) < 5` ?

Reply:
We can't get here without

```go
if len(doc.RecentSequences) >= kMaxRecentSequences {
```

`maxToCompact` can't be negative